subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/RepositoryContentDescriptor.java

```diff
@@ -104,15 +105,15 @@ public interface RepositoryContentDescriptor extends InclusiveRepositoryContentD
      *
      * @param configurationNames the names of the configurations the repository will be used for
      */
-    void onlyForConfigurations(String... configurationNames);
+    void onlyForConfigurations(@Nullable String... configurationNames);
 
     /**
      * Declares that this repository should not be used for a specific
      * set of configurations. Defaults to any configuration
      *
      * @param configurationNames the names of the configurations the repository will not be used for
      */
-    void notForConfigurations(String... configurationNames);
+    void notForConfigurations(@Nullable String... configurationNames);
```

Comment:
Same here.

Reply:
The implementation does accept null, so passing null was working in 8.x.